### PR TITLE
Add an e2e (host-independent) test demo to testing_app

### LINKED ISSUE #512: Add an e2e test case to testing_app
Currently this app doesn't include e2e test.

### PR DESCRIPTION
**Pull request by CareF**
As I offerred in  https://github.com/flutter/samples/pull/500#discussion_r462694502 .
Resolves #512
@abd99

### CODE REVIEW

File: testing_app/test/perf_test_e2e.dart [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This file duplicate the behavior of `test_driver/perf_test.dart` but uses
+// e2e package to implement a host-independent test.
+
+import 'package:flutter/material.dart';
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:e2e/e2e.dart';
```
Comment (abd99) [inline testing_app/test/perf_test_e2e.dart:11]: Please sort the imports here. We have a convention on keeping the imports sorted in this repo.

Comment (CareF) [reply testing_app/test/perf_test_e2e.dart:11]: Done

**Comment by CareF** [comment]
So after some discussion for the `e2e` package, part of the feature we need from e2e is moved to `flutter_test`, because `e2e` needs to be compatible with the stable SDK version. I'm not sure the SDK version policy in this repo, so I'm copying related piece of code (`test/e2e_utils.dart`) to the PR, which should land to the flutter SDK and/or e2e when current master goes to stable, but feel free to close this PR if that means we shouldn't include this here.

File: testing_app/test/perf_test_e2e.dart [hunk 2]
```diff
@@ -0,0 +1,79 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This file duplicate the behavior of `test_driver/perf_test.dart` but uses
```
Comment (RedBrogdon) [inline testing_app/test/perf_test_e2e.dart:5]: nit: "This file duplicates the behavior of `test_driver/perf_test.dart`, but uses the e2e package to implement a host-independent test."

File: testing_app/test/perf_test_e2e.dart [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:testing_app/main.dart' as app;
+
+import 'e2e_utils.dart';
+
+void main() {
+  final binding =
+      E2EWidgetsFlutterBinding.ensureInitialized() as E2EWidgetsFlutterBinding;
+  // fullyLive frame policy simulates the way Flutter response to animations.
```
Comment (RedBrogdon) [inline testing_app/test/perf_test_e2e.dart:18]: "The fullyLive frame policy simulates the way Flutter responds to animations"?

File: testing_app/test_driver/e2e_test.dart [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:e2e/e2e_driver.dart' as driver;
+
+Future<void> main() => driver.e2eDriver(responseDataCallback: (data) async {
+      await driver.writeResponseData(
+        data['scrolling'] as Map<String, dynamic>,
```
Comment (RedBrogdon) [inline testing_app/test_driver/e2e_test.dart:11]: Are these values also written to stdout, so they're visible in CI logs?

File: testing_app/pubspec.lock [hunk 5]
```diff
@@ -503,5 +503,5 @@ packages:
     source: hosted
     version: "2.2.1"
 sdks:
-  dart: ">=2.7.0 <3.0.0"
-  flutter: ">=1.16.0"
+  dart: ">=2.10.0-0.0.dev <2.10.0"
```
Comment (RedBrogdon) [inline testing_app/pubspec.lock:506]: What channel were you using when you updated this file?

File: testing_app/test_driver/e2e_test.dart (hunk 4 above)
Comment (CareF) [reply testing_app/test_driver/e2e_test.dart:11]: By default no, but I can add a print.

File: testing_app/pubspec.lock (hunk 5 above)
Comment (CareF) [reply testing_app/pubspec.lock:506]: master. I pushed a new commit using stable.

File: testing_app/test/perf_test_e2e.dart (hunk 3 above)
Comment (CareF) [reply testing_app/test/perf_test_e2e.dart:18]: Done

File: testing_app/test/perf_test_e2e.dart (hunk 2 above)
Comment (CareF) [reply testing_app/test/perf_test_e2e.dart]: Done.

File: testing_app/test_driver/e2e_test.dart (hunk 4 above)
Comment (CareF) [reply testing_app/test_driver/e2e_test.dart:12]: Done

**Comment by CareF** [comment]
> We have the ability to run tests in this repo via Firebase Test Lab. Would it make sense to wire these new tests up to run there?

It sure can, but for now we still don't have a good solution about how to report performance metrics for tests on firebase test lab. But since this is not part of Framework performance test set, I would assume it's fine to ignore it for now. 

I'll try a script but I'm not very familiar with the CI system here, so any advice would be very appreciated. 
@RedBrogdon

File: testing_app/pubspec.lock (hunk 5 above)
Comment (CareF) [reply testing_app/pubspec.lock]: Actually I wonder why `pubspec.lock` is not ignored in git in this repo?

**Pull request merged**